### PR TITLE
Added acceptance test for Dhcp options in various resources that supports it

### PR DESCRIPTION
**Pull request by AnilGadiyarHJ**
api_subnet_resource
api_range_resource
api_ip_space_resource
api_fixed_address_resource
api_address_block_resource
api_server
https://infoblox.atlassian.net/browse/DDIA-212

### CODE REVIEW

File: docs/resources/dhcp_fixed_address.md [hunk 1]
```diff
@@ -69,6 +69,14 @@ resource "bloxone_dhcp_fixed_address" "example_fixed_address_na" {
   tags = {
     location : "site1"
   }
+  dhcp_options = [
+    {
+      description  = "Option 1"
+      option_code  = "234"
+      option_value = "true"
+      type         = "boolean"
+    }
```
Comment (mathewab) [inline docs/resources/dhcp_fixed_address.md:78]: I don't think this will work. `description` is not a field part of dhcp_options. Also `option_code` is an UUID for the dhcp_option_code object and not just the code. And `type` here is either `option` or `group`.

File: internal/service/ipam/model_ipamsvc_range.go [hunk 2]
```diff
@@ -95,6 +96,9 @@ var IpamsvcRangeResourceSchemaAttributes = map[string]schema.Attribute{
 		},
 		Optional:            true,
 		MarkdownDescription: "The list of DHCP options. May be either a specific option or a group of options.",
+		PlanModifiers: []planmodifier.List{
+			listplanmodifier.RequiresReplaceIfConfigured(),
+		},
```
Comment (mathewab) [inline internal/service/ipam/model_ipamsvc_range.go:101]: Why did you add this? This will basically delete the range and recreate it every time the `dhcp_options` is modified.

Comment (AnilGadiyarHJ) [reply internal/service/ipam/model_ipamsvc_range.go:101]: DHCP option has 2 types, we can either create via option & group. we have added both in function steps and the update was not working when steps were creating DHCP option via option type and then update group type.

File: docs/resources/dhcp_fixed_address.md (hunk 1 above)
Comment (AnilGadiyarHJ) [reply docs/resources/dhcp_fixed_address.md]: done

File: docs/resources/dhcp_fixed_address.md [hunk 3]
```diff
@@ -58,6 +58,13 @@ resource "bloxone_dhcp_fixed_address" "example_fixed_address" {
   depends_on = [bloxone_ipam_subnet.example]
 }
 
+resource "bloxone_dhcp_option_code" "option_code" {
+  code         = 250
+  name         = "example_option_code"
+  option_space = bloxone_dhcp_option_space.option_space.id
+  type         = "int32"
+}
+
```
Comment (mathewab) [inline docs/resources/dhcp_fixed_address.md:67]: I would rather use a data source to get an existing Option Code (like for ex. option 6). This is a more common scenario.

File: internal/service/ipam/api_subnet_resource_test.go [hunk 4]
```diff
@@ -1059,6 +1090,77 @@ resource "bloxone_ipam_subnet" "test_ddns_conflict_resolution_mode" {
...
+    code = "234"
+    name = "test_dhcp_option_code"
+    option_space = bloxone_dhcp_option_space.test.id
+    type = "boolean"
+}
+`
+	return strings.Join([]string{testAccOptionSpace("test_option_space", "ip4"), config}, "")
+}
+
+func testAccOptionBasicConfigV6() string {
```
Comment (mathewab) [inline internal/service/ipam/api_subnet_resource_test.go:1148]:
```suggestion
func testAccBaseWithV6OptionSpaceAndCode() string {
```

File: internal/service/ipam/api_subnet_resource_test.go [hunk 5]
```diff
@@ -1059,6 +1090,77 @@ resource "bloxone_ipam_subnet" "test_ddns_conflict_resolution_mode" {
...
+       group = bloxone_dhcp_option_group.test.id
+      }
+    ]
+}
+`, address, cidr, name, type_)
+
+	return strings.Join([]string{testAccBaseWithIPSpace(), testAccOptionBasicConfig(), config}, "")
+}
+
+func testAccOptionBasicConfig() string {
```
Comment (mathewab) [inline internal/service/ipam/api_subnet_resource_test.go:1132]:
```suggestion
func testAccBaseWithOptionSpaceAndCode() string {
```

Comment (mathewab) [reply internal/service/ipam/api_subnet_resource_test.go:1132]: If this function is already used in the basic test, do not reuse it, create another function with the above name.

File: docs/resources/dhcp_fixed_address.md (hunk 3 above)
Comment (AnilGadiyarHJ) [reply docs/resources/dhcp_fixed_address.md]: done

File: internal/service/ipam/api_subnet_resource_test.go (hunk 4 above)
Comment (AnilGadiyarHJ) [reply internal/service/ipam/api_subnet_resource_test.go]: done

File: internal/service/ipam/api_subnet_resource_test.go (hunk 5 above)
Comment (AnilGadiyarHJ) [reply internal/service/ipam/api_subnet_resource_test.go]: This is a new function created. renamed it

File: internal/service/ipam/model_ipamsvc_address_block.go [hunk 6]
```diff
@@ -238,6 +239,9 @@ var IpamsvcAddressBlockResourceSchemaAttributes = map[string]schema.Attribute{
 		},
 		Optional:            true,
 		MarkdownDescription: "The list of DHCP options for the address block. May be either a specific option or a group of options.",
+		PlanModifiers: []planmodifier.List{
+			listplanmodifier.RequiresReplaceIfConfigured(),
+		},
```
Comment (mathewab) [inline internal/service/ipam/model_ipamsvc_address_block.go:244]: This one still needs to be removed.

File: internal/service/ipam/model_ipamsvc_fixed_address.go [hunk 7]
```diff
@@ -2,6 +2,7 @@ package ipam
 
 import (
 	"context"
+
```
Comment (mathewab) [inline internal/service/ipam/model_ipamsvc_fixed_address.go:5]: line break not needed here

File: internal/service/ipam/model_ipamsvc_subnet.go [hunk 8]
```diff
@@ -2,6 +2,7 @@ package ipam
 
 import (
 	"context"
+
```
Comment (mathewab) [inline internal/service/ipam/model_ipamsvc_subnet.go:5]: line break not needed here

File: docs/resources/dhcp_fixed_address.md [hunk 9]
```diff
@@ -55,6 +62,14 @@ resource "bloxone_dhcp_fixed_address" "example_fixed_address" {
   tags = {
     location = "site1"
   }
+  //dhcp options
+  dhcp_options = [
+    {
+      option_code  = data.bloxone_dhcp_option_codes.option_code.results.0.id
+      option_value = "1.1.1.1"
```
Comment (mathewab) [inline docs/resources/dhcp_fixed_address.md:69]: can you change all `1.1.1.1` to `10.0.0.1` instead.

File: internal/service/ipam/model_ipamsvc_address_block.go (hunk 6 above)
Comment (AnilGadiyarHJ) [reply internal/service/ipam/model_ipamsvc_address_block.go]: done

File: internal/service/ipam/model_ipamsvc_fixed_address.go (hunk 7 above)
Comment (AnilGadiyarHJ) [reply internal/service/ipam/model_ipamsvc_fixed_address.go]: done

File: internal/service/ipam/model_ipamsvc_subnet.go (hunk 8 above)
Comment (AnilGadiyarHJ) [reply internal/service/ipam/model_ipamsvc_subnet.go]: done

File: docs/resources/dhcp_fixed_address.md (hunk 9 above)
Comment (AnilGadiyarHJ) [reply docs/resources/dhcp_fixed_address.md]: done

**Review by mathewab — APPROVED** [review]
